### PR TITLE
DAT-3294 add infobox-hero-image context handling

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/DAT-3294

Do not merge without mercury part: https://github.com/Wikia/mercury/pull/1447

### CODE REVIEW

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 1]
```diff
@@ -134,13 +134,14 @@ public static function onGalleryBeforeProduceHTML( $data, &$out ){
 		return true;
 	}
 
-	public static function onPortableInfoboxNodeImageGetData( $title, &$ref, $caption ) {
-
+	public static function onPortableInfoboxNodeImageGetData( $data, &$ref ) {
 		wfProfileIn( __METHOD__ );
+
+		$title = Title::newFromText( $data[ 'name' ] );
```
Comment (rogatty) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: Unnecessary spaces:
`$arr = [ $one, $two ]`
but
`$var = $arr['index']`

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 2]
```diff
@@ -134,13 +134,14 @@ public static function onGalleryBeforeProduceHTML( $data, &$out ){
 		return true;
 	}
 
-	public static function onPortableInfoboxNodeImageGetData( $title, &$ref, $caption ) {
-
+	public static function onPortableInfoboxNodeImageGetData( $data, &$ref ) {
```
Comment (RafLeszczynski) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: Consider changing name both for this method and the hook itself. This hook is run while extending image data in render service not while getting data from image node

File: extensions/wikia/ArticleAsJson/ArticleAsJson.class.php [hunk 3]
```diff
@@ -134,13 +134,14 @@ public static function onGalleryBeforeProduceHTML( $data, &$out ){
...
 		wfProfileIn( __METHOD__ );
+
+		$title = Title::newFromText( $data['name'] );
 		if ( $title ) {
 			$details = WikiaFileHelper::getMediaDetail( $title, self::$mediaDetailConfig );
-			$details['context'] = self::MEDIA_CONTEXT_INFOBOX;
-			self::$media[] = self::createMediaObject( $details, $title->getText(), $caption );
+			$details['context'] = isset( $data['context'] ) ? $data['context'] : self::MEDIA_CONTEXT_INFOBOX;
```
Comment (RafLeszczynski) [inline extensions/wikia/ArticleAsJson/ArticleAsJson.class.php]: context for here image in infobox is set in infobox extension and context for regular infobox image is set here. Please consider setting both in single place.

**Comment by RafLeszczynski** [comment]
LGTM

File: extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php [hunk 4]
```diff
@@ -149,6 +152,7 @@ private function renderItem( $type, array $data ) {
 		$helper = new PortableInfoboxRenderServiceHelper();
 
 		if ( $type === 'image' ) {
+			$data[ 'image' ][ 'context' ] = self::MEDIA_CONTEXT_INFOBOX;
```
Comment (inez) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php:155]: @dianafa Are you sure that it should be:
`$data[ 'image' ][ 'context' ] = self::MEDIA_CONTEXT_INFOBOX;`
and not:
`$data[ 'context' ] = self::MEDIA_CONTEXT_INFOBOX;`
?

Comment (dianafa) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php:155]: good eye! :+1:

Comment (dianafa) [inline extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php:155]: and a follow-up PR: https://github.com/Wikia/app/pull/8707

**Pull request merged**